packages/medusa/src/api/routes/admin/products/update-variant.ts

```diff
@@ -105,56 +119,136 @@ import { defaultFields, defaultRelations } from "./"
-        })
-        .xor("region_id", "currency_code")
-    ),
-    options: Validator.array().items({
-      option_id: Validator.string().required(),
-      value: Validator.alternatives(
-        Validator.string(),
-        Validator.number()
-      ).required(),
-    }),
-    inventory_quantity: Validator.number().optional(),
-    allow_backorder: Validator.boolean().optional(),
-    manage_inventory: Validator.boolean().optional(),
-    weight: Validator.number().allow(null).optional(),
-    length: Validator.number().allow(null).optional(),
-    height: Validator.number().allow(null).optional(),
-    width: Validator.number().allow(null).optional(),
-    hs_code: Validator.string().optional().allow(null, ""),
-    origin_country: Validator.string().optional().allow(null, ""),
-    mid_code: Validator.string().optional().allow(null, ""),
-    material: Validator.string().optional().allow(null, ""),
-    metadata: Validator.object().optional(),
-  })
 
-  const { value, error } = schema.validate(req.body)
-  if (error) {
-    throw new MedusaError(MedusaError.Types.INVALID_DATA, error.details)
-  }
+  const validated = await validator(
+    AdminPostProductsProductVariantsVariantReq,
+    req.body
+  )
 
-  const productService = req.scope.resolve("productService")
-  const productVariantService = req.scope.resolve("productVariantService")
+  const productService: ProductService = req.scope.resolve("productService")
+  const productVariantService: ProductVariantService = req.scope.resolve(
+    "productVariantService"
+  )
 
-  await productVariantService.update(variant_id, value)
+  await productVariantService.update(variant_id, {
+    product_id: id,
+    ...validated,
+  })
 
   const product = await productService.retrieve(id, {
-    select: defaultFields,
-    relations: defaultRelations,
+    select: defaultAdminProductFields,
+    relations: defaultAdminProductRelations,
   })
 
   res.json({ product })
 }
+
+class ProductVariantOptionReq {
+  @IsString()
+  value: string
+
+  @IsString()
+  option_id: string
+}
+
+class ProductVariantPricesReq {
+  @Validate(XorConstraint, ["currency_code"])
+  region_id?: string
+
+  @Validate(XorConstraint, ["region_id"])
+  currency_code?: string
+
+  @IsNumber()
+  @Type(() => Number)
+  amount: number
+
+  @IsOptional()
+  @IsNumber()
+  @Type(() => Number)
+  sale_amount: number
+}
+
+class AdminPostProductsProductVariantsVariantReq {
+  @IsString()
+  @IsOptional()
+  title?: string
+
+  @IsString()
+  @IsOptional()
+  sku?: string
+
+  @IsString()
+  @IsOptional()
+  ean?: string
+
+  @IsString()
+  @IsOptional()
+  upc?: string
+
+  @IsString()
+  @IsOptional()
+  barcode?: string
+
+  @IsString()
+  @IsOptional()
+  hs_code?: string
+
+  @IsNumber()
+  @IsOptional()
+  @Type(() => Number)
+  inventory_quantity: number
+
+  @IsBoolean()
+  @IsOptional()
+  allow_backorder?: boolean
+
+  @IsBoolean()
+  @IsOptional()
+  manage_inventory?: boolean
```

Comment:
Missing type transform? `@Type(() => Boolean)`

Reply:
Not on properties in `req.body`, they are correctly passed, Type annotations are only required for simple types in query params